tw-gaffer-jta/src/main/java/com/transferwise/common/gaffer/jdbc/GafferJtaDataSource.java

```diff
@@ -40,6 +43,16 @@ public class DataSourceImpl extends DataSourceWrapper implements DataSourceManag
   private final AtomicLong autoCommitSwitchingsCount = new AtomicLong();
   private final AtomicLong closedConnectionsCount = new AtomicLong();
 
+  public GafferJtaDataSource() {
```

Comment:
Since we allow this constructor, then the setTargetDatasource should probably also call the `DatasourceProxyUtils.tieTogether()`

Reply:
This would create a stack overflow, because DatasourceProxyUtils itself can also call this method.